File: Hw1/Hw1.java
import java.util.Scanner;
import java.util.Arrays;


public class Hw1 {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		System.out.println(""+args[0]);									
		String str = args[0];											//turn argument to string


		str = str.replaceAll("[^a-zA-Z]","");							//only letters can be left
		str = str.toLowerCase();										//transfer uppercase to lowercase
		
		int cnt=0;
		char[] array = str.toCharArray();								//transfer string to array
		int len = array.length;

		for( int i=0; i<array.length/2; i++,len-- ) {					//check whether front and back sides are same
			if( array[i] == array[len-1] )
				cnt++;	
			else
				break;
		}
		
		if( cnt == array.length/2 )
			System.out.println("this string is palindrome");
		else
			System.out.println("this string is not palindrome");	
	}
}

File: Hw1/Hw1.cpp
#include <iostream>
#include <locale>
#include <string>

using namespace std;
int main(int argc , char *argv[])
{

		string str(argv[1]);										// turn argument to string
		cout<< str << endl;	
	
		int len = str.size() , j=0 , cnt=0 , i;
		char array[len+1];	

		for( i=0; i<len; i++ ) {										
			if( isalpha(str[i]) ) {									//find letters only, and put it inot an array
				str[i] = tolower(str[i]);							//transfer uppercase to lowercase
				array[j++] = str[i];
			}
		}
		len = j;
		for( i=0; i<len/2; i++,j-- ) {								//create a loop to check whether the string is palindrome
			if( array[i] == array[j-1] )
				cnt++;														
		}
		if( cnt == len/2 )											//if cnt equals len/2(odd or even doesn't matter,since "len" is integer) , it means the string is palindorme		
			cout << "This String is Palindrome" <<endl;
		else
			cout << "This String is not Palindrome" << endl;	
}
